builtin/providers/aws/resource_aws_alb_target_group_attachment.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAlbTargetGroupAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAlbAttachmentCreate,
+		Read:   resourceAwsAlbAttachmentRead,
+		Delete: resourceAwsAlbAttachmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"target_group_arn": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"target_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"port": {
+				Type:     schema.TypeInt,
+				ForceNew: true,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAlbAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+
+	params := &elbv2.RegisterTargetsInput{
+		TargetGroupArn: aws.String(d.Get("target_group_arn").(string)),
+		Targets: []*elbv2.TargetDescription{
+			{
+				Id:   aws.String(d.Get("target_id").(string)),
+				Port: aws.Int64(int64(d.Get("port").(int))),
+			},
+		},
+	}
+
+	log.Printf("[INFO] Registering Target %s (%d) with Target Group %s", d.Get("target_id").(string),
+		d.Get("port").(int), d.Get("target_group_arn").(string))
+
+	_, err := elbconn.RegisterTargets(params)
+	if err != nil {
+		return errwrap.Wrapf("Error registering targets with target group: {{err}}", err)
+	}
+
+	d.SetId(resource.PrefixedUniqueId(fmt.Sprintf("%s-", d.Get("target_group_arn"))))
+
+	return nil
```

Comment:
I don't think there's any value in calling Read here since there is nothing computed (or in fact set in the Read - whose only function is to ensure that the attachment still exists).